stock_available_mrp/models/product_product.py

```diff
@@ -104,12 +103,20 @@ def _compute_available_quantities_dict(self):
                 # Find the lowest quantity we can make with the stock at hand
                 components_potential_qty = min(
                     [component_qties[component.id][
-                        stock_available_mrp_based_on] // need
+                        stock_available_mrp_based_on] / need
                      for component, need in component_needs.items()]
                 )
 
-                potential_qty = (product.bom_id.product_qty *
-                                 components_potential_qty)
+                bom_id = product.bom_id
+                potential_qty = (bom_id.product_qty * components_potential_qty)
+
+                # We want to respect the rounding factor of the potential_qty
+                # Rounding down as we want to be pesimistic.
+                potential_qty = bom_id.product_uom_id._compute_quantity(
+                    potential_qty,
+                    product.bom_id.product_tmpl_id.uom_id,
+                    rounding_method='DOWN'
```

Comment:
@tschanzt Just a question about this. The standard option is 'UP'. So, why rounding down as it could be weird for users if behaviour is not the same on other places?

Reply:
@rousseldenis The point of it is to be pesimistic about how much you can deliver. The original problem was, that we didn't respect the rounding of the uom. E.G. if you want to make a skateboard (requiring the board and 4 wheels) and you had the board and 2 wheels it would return 0.5 instead of 0 so i added this to apply the rounding, but if we round up we then get 1 as the potential quantity which isn't what we can produce.

Reply:
Makes sense ! So, let's go